src/terraform.ts

```diff
@@ -44,6 +56,39 @@ interface ModuleProvidersResponse {
 }
 /* eslint-enable @typescript-eslint/naming-convention */
 
+export async function getTerraformVersion(
+  moduleUri: vscode.Uri,
+  client: LanguageClient,
+  reporter: TelemetryReporter,
+): Promise<void> {
+  try {
+    const moduleDir = Utils.dirname(moduleUri);
+
+    const response = await terraformVersion(moduleDir.toString(), client, reporter);
+    tfStatus.setTerraformVersion(response.discovered_version);
```

Comment:
I'm not sure if this is the place where I would expect the modification of the status bar item. Most of the other functions in this file make an API call and return the result.

Maybe this line belongs in the `TerraformVersionFeature`?